src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -432,10 +440,7 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
       else {
         val table0 = table
         val default = defaultValue
-        if(contentSize + 1 >= threshold) growTable(table.length * 2)
-        // Avoid recomputing index if the `defaultValue()` or new element hasn't triggered a table resize.
-        val newIdx = if (table0 eq table) idx else index(hash)
-        put0(key, default, false, hash, newIdx)
+        put0(key, default, false, hash, idx)
```

Comment:
You can't remove these checks. `defaultSize` may have triggered a resize, in which case `idx` will be invalid. I wouldn't remove the explicit resize, either. You already know that the table will grow by 1 (because the key was not found) so you may as well do it here. The only downside is the double `if(contentSize + 1 >= threshold)` check, both here and in `put0`. But the second one will always yield `false`. There's no risk of resizing twice. OTOH, if you remove it here, you could end up recomputing the index unnecessarily in corner cases.

Reply:
👍  thanks

Reply:
On second thought, this corner case is so unlikely that avoiding the extra check may be better. The difference will be small either way.